[12.0][ADD] slow_statement_logger

sbidoul opened the pull request:
This module lets you log slow SQL statements. It is useful when you don't have
access to the PostgreSQL log or when it is difficult to correlate the
PostgreSQL log with the Odoo log.

sbidoul committed:
[ADD] slow_statement_logger

ThomasBinsfeld commented on slow_statement_logger/__manifest__.py:
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "Slow SQL Statement Logger",
+    "summary": "Log slow SQL statments",
```
```suggestion
    "summary": "Log slow SQL statements",
```

ThomasBinsfeld commented on slow_statement_logger/readme/USAGE.rst:
```diff
@@ -0,0 +1,10 @@
+Add ``slow_statment_logger`` to Odoo's ``--load`` command line option, or to the
```
```suggestion
Add ``slow_statement_logger`` to Odoo's ``--load`` command line option, or to the
```

ThomasBinsfeld commented on slow_statement_logger/readme/USAGE.rst:
```diff
@@ -0,0 +1,10 @@
+Add a ``log_min_duration_statement`` entry in the ``options`` section of your
+Odoo configuration file. Statements running at least this number of
+milliseconds will be logged with a *warning* level in the
+``odoo.addons.slow_statment_logger`` handler. ``0`` means all statements will be
```
```suggestion
``odoo.addons.slow_statement_logger`` handler. ``0`` means all statements will be
```

sbidoul committed:
slow_statement_logger: log with debug level

This is effectively a debugging information and logging
with a warning level would create too much noise in
log capture systems such as Sentry.

sbidoul commented:
After further thinking I decided to log with `debug` level.

sbejaoui commented:
Very useful, Thank you (y)

lmignon commented:
/ocabot merge
